app/Http/Filters/Organisation/HasSocialMediasFilter.php

```diff
@@ -15,10 +18,28 @@ class HasSocialMediasFilter implements Filter
      */
     public function __invoke(Builder $query, $value, string $property): Builder
     {
-        $hasServices = (bool)$value;
+        switch ($value) {
+            case 'any':
+                $query = $query->whereHas('socialMedias');
+                break;
+            case 'none':
+                $query = $query->whereDoesntHave('socialMedias');
+                break;
+            case SocialMedia::TYPE_FACEBOOK:
+            case SocialMedia::TYPE_TWITTER:
+            case SocialMedia::TYPE_INSTAGRAM:
+            case SocialMedia::TYPE_YOUTUBE:
+            case SocialMedia::TYPE_OTHER:
+                $query = $query->whereExists(function ($query) use ($value) {
+                    $query->select(DB::raw(1))
+                        ->from('social_medias')
+                        ->whereRaw('organisations.id = social_medias.sociable_id')
+                        ->whereRaw('social_medias.sociable_type = ?', [table(Organisation::class)])
```

Comment:
The morph map overrides the default:
```suggestion
                        ->whereRaw('social_medias.sociable_type = ?', ['organisations'])
```

https://github.com/Healthy-London-Partnership/api/blob/bef6b85ed66c4c0b1cbedb3829f7c15848d735ff/app/Providers/ModelServiceProvider.php#L56-L61

Reply:
Done.
Sorry I assumed the table function was designed to work with the morph map. Having had a look inside I can see it just calculates a standard table name. I'll take this into account in future.

Reply:
@appsol Yeah, it does. Sorry I completely missed the use of `table()` - that would have worked. I'd still avoid the use in the case though, as morph map just happens to be using the same name as the tables, but there's nothing stopping that from changing in the future.